internal/services/blackwhitelist_test.go

```diff
@@ -86,151 +84,205 @@ func TestGenerateProjects(t *testing.T) {
 			Project:     "native",
 			Environment: "development",
 		},
+		{
```

Comment:
the tests are hard to read. could you use a triple slice in expected, containing created, deleted, and ignored projects, so we can loop over tests with very explicit cases?